src/main/kotlin/app/revanced/patches/youtube/misc/fix/playback/SpoofSignaturePatch.kt

```diff
@@ -38,29 +41,60 @@ object SpoofSignaturePatch : BytecodePatch(
+            PreferenceScreen(
+                key = "revanced_spoof_signature_verification",
+                title = StringResource(
+                    "revanced_spoof_signature_verification_title",
+                    "Spoof app signature"
+                ),
+                preferences = listOf(
+                    SwitchPreference(
+                        "revanced_spoof_signature_verification_enabled",
+                        StringResource("revanced_spoof_signature_verification_enabled_title", "Spoof app signature"),
+                        StringResource(
+                            "revanced_spoof_signature_verification_enabled_summary_on",
+                            "App signature spoofed\\n\\n"
+                                    + "Side effects include:\\n"
+                                    + "• No ambient mode\\n"
+                                    + "• Videos cannot be downloaded\\n"
+                                    + "• Low quality seekbar thumbnails (No thumbnails for age restricted or paid videos)"
+                        ),
+                        StringResource(
+                            "revanced_spoof_signature_verification_enabled_summary_off",
+                            "App signature not spoofed\\n\\nVideo playback may not work"
+                        ),
+                        StringResource(
+                            "revanced_spoof_signature_verification_enabled_user_dialog_message",
+                            "Turning off this setting will cause video playback issues."
+                        )
+                    ),
+                    SwitchPreference(
+                        "revanced_spoof_signature_in_feed_enabled",
+                        StringResource("revanced_spoof_signature_in_feed_enabled_title", "Spoof app signature in feed"),
+                        StringResource(
+                            "revanced_spoof_signature_in_feed_enabled_summary_on",
+                            "App signature spoofed\\n\\n"
+                                    + "Side effects include:\\n"
+                                    + "• Feed videos are missing subtitles\\n"
+                                    + "• Automatically played feed videos will show up in your watch history"
+                        ),
+                        StringResource(
+                            "revanced_spoof_signature_in_feed_enabled_summary_off",
+                            "App signature not spoofed for feed videos\n\n"
+                                    + "Feed videos will play for less than 1 minute before encountering playback issues"
+                        )
+                    )
                 )
-            }
-        } ?: throw ProtobufParameterBuilderFingerprint.exception
+            )
+        )
+
+        // Hook the player parameters.
+        VideoIdPatch.injectProtoBufferHook("$INTEGRATIONS_CLASS_DESCRIPTOR->spoofParameter(Ljava/lang/String;)Ljava/lang/String;")
```

Comment:
This does not make sense to me. Why does `VideoIdPatch` hook the player parameter? `VideoIdPatch` should only hook video ids